wweval/R/get_input_data.R

```diff
@@ -43,51 +43,48 @@ get_input_ww_data <- function(forecast_date_i,
   ww_data_path <- file.path(ww_data_dir, paste0(date_to_pull, ".csv"))
   raw_nwss_data <- readr::read_csv(ww_data_path, show_col_types = FALSE)
 
-  # Use package functions to subset NWSS data
-  ww_data <- raw_nwss_data |>
-    wweval::init_subset_nwss_data()
+  # Use wweval functions to subset NWSS data and format for wwinference package
+  all_ww_data <- raw_nwss_data |>
+    clean_and_filter_nwss_data()
   # Get the data corresponding to the scenario
   list_of_site_ids <- get_scenario_site_ids(
-    ww_data,
+    all_ww_data,
     scenario_i,
     scenario_dir
   )
-  subsetted_ww_data <- ww_data |>
+  subsetted_ww_data <- all_ww_data |>
     dplyr::filter(wwtp_name %in% !!list_of_site_ids)
 
-  ww <- subsetted_ww_data |>
+  ww_data_pkg <- subsetted_ww_data |>
     clean_ww_data() |>
     filter(
       location %in% c(!!location_i),
       date >= lubridate::ymd(!!last_hosp_data_date) -
         lubridate::days(!!calibration_time) + lubridate::days(1)
     )
 
-  # Get extra columns that identify wastewater outliers
-  ww_w_outliers <- wwinference::flag_ww_outliers(ww) |>
-    select(
-      date, location, ww, site, lab, lab_wwtp_unique_id,
-      ww_pop, below_LOD, lod_sewage, flag_as_ww_outlier
-    )
-  # If more than one location, than this data isn't being used for fitting
-  # And we don't wanto generate these
-  if (length(location_i) == 1) {
-    site_map <- ww_w_outliers |>
-      distinct(site) |>
-      mutate(site_index = row_number())
-    site_lab_map <- ww_w_outliers |>
-      distinct(lab_wwtp_unique_id) |>
-      mutate(lab_site_index = row_number())
+  deduplicated_data <- ww_data_pkg |>
+    dplyr::group_by(lab, site, date) |>
+    dplyr::summarize(
```

Comment:
good opportunity to use `accross`

Reply:
I have never used it, can you provide an example?

Reply:
`summarize(across(c(log_genome_copies_per_ml, log_lod, site_pop), mean))`

is exactly equivalent to your 

```r
dplyr::summarize(
      log_genome_copies_per_ml = mean(log_genome_copies_per_ml),
      log_lod = mean(log_lod),
      site_pop = mean(site_pop)
)
```